draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,551 @@ author:
+  (and thus not be returned).
+
+
+### Example Proxy-DNS-SVCB response
+
+As an example, assume that the server resolved the SVCB DNS query name
+"example.com" and RR type 65 (HTTPS) as follows:
 
-Each SvcParam that matches a key requested by the client is a parameters with a key
-that is the character "p" followed by the numeric version of the SvcParamKey. For example,
-the ALPN SvcParamKey, with the numeric value 1, would have a parameter key "p1". The value
-of each parameter MUST be an sf-binary item that contains the bytes of the SvcParamValue.
+~~~ diagram
+   example.com.     7200 IN HTTPS 0 foo.svc.example.net.
+   foo.svc.example.net 1800 IN CNAME svc.example.net.
+   svc.example.net. 3600 IN HTTPS 1 svc2.example.net. alpn=h2,h3 echconfig="123..."
+   svc.example.net. 3600 IN HTTPS 2 . alpn=h2 echconfig="abc..."
+~~~
+
+A successful CONNECT response would include the following headers, if the client requested both
+"alpn" and "echconfig":
+
+~~~ example
+HEADERS
+:method = CONNECT
+:status = 200
+proxy-dns-svcb = "svc2.example.net.";priority=1;ttl=1800;key1=:aDIsaDM=:;key5=:MTIzLi4u:,
+                 "svc.example.net.";priority=2;ttl=1800;key1=:aDI=:;key5=:YWJjLi4u:
+~~~
+
+
+## Proxy-DNS-Used Response Header Field {#proxy-dns-used}
+
+A proxy server that receives a request with "Proxy-DNS-Request" with MAY respond with
+the Structured Header "Proxy-DNS-Used" response header fields. The value of
+"Proxy-DNS-Used" MUST be an sf-list whose members are sf-string, each of
+which MUST contain parameters.
+
+~~~ abnf
+Proxy-DNS-Used = sf-list
+~~~
 
-Proxy servers MUST NOT include the "DNS-SVCB-Params" response header field if the
-corresponding request did not include a "DNS-SVCB-Keys". Servers MAY include
-specific SvcParamKey values that were not requested. Specifically, servers SHOULD include
-the "mandatory" parameter if present, which would be presented as "p0", along with any
-parameters that are defined as mandatory for that record.
+The intent of this header is to provide clients with enough
+information to implement Section 5 of {{SVCB}}.
 
-As an example, assume that the server received the following "svc.example.com" SVCB records:
+Proxy servers MUST NOT include the "Proxy-DNS-Used" response header field if the
+corresponding request did not include a "Proxy-DNS-Request" or if its "u"
+parameter had the value sf-false.
+
+Each list member is an sf-string that represents either the IP address
+(IPv4 or IPv6) that the proxy connected to for the CONNECT* request,
+or the value of a DNS CNAME on the path of aliases involved in
+resolving the hostname connected to by the proxy.
+
+The last member in the list MUST have an item value that is the string
+representation of the IPv6 or IPv4 address that represents the
+destination of the CONNECT* from the proxy.  If this is an IPv6
+address, it MUST follow the canonical string form from {{!RFC5952}}.
+
+*TODO*: Is there an IPv4 canonical form RFC we should reference here?
```

Comment:
We might use `host` out of RFC 3986, since it specifies IPv4, IPv6, and hostname rendering.